vars/getDeploymentResourcesK8s.groovy

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/groovy
+
+def call(body) {
+    // evaluate the body block, and collect configuration into the object
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    def expose = config.exposeApp ?: 'true'
+    def requestCPU = config.resourceRequestCPU ?: '0'
+    def requestMemory = config.resourceRequestMemory ?: '0'
+    def limitCPU = config.resourceLimitMemory ?: '0'
+    def limitMemory = config.resourceLimitMemory ?: '0'
+    def ingressClass = config.ingressClass ?: 'unknown'
+    def readinessProbePath = config.readinessProbePath ?: "/"
+    def livenessProbePath = config.livenessProbePath ?: "/"
+
+    def yaml
+
+    def fabric8Registry = ''
+    if (env.FABRIC8_DOCKER_REGISTRY_SERVICE_HOST){
+        fabric8Registry = env.FABRIC8_DOCKER_REGISTRY_SERVICE_HOST+':'+env.FABRIC8_DOCKER_REGISTRY_SERVICE_PORT
+    }
+
+    def list = """
+---
+apiVersion: v1
+kind: List
+items:
+"""
+
+def service = """
```

Comment:
I think these should be in separate files as templates and we can add placeholders for these values. We can then just read the template and replace those placeholders with the actual values. I'm not sure if this could be done in this case, but it would clean up these files

Reply:
i have been thinking that we need a more clean solution for this; e.g. use Helm charts as they are for this purpose with support of templatizing etc. but for now we can go this way.

but very good you pointed it out!